recipes/opentdf-client/all/conanfile.py

```diff
@@ -78,6 +79,17 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def configure(self):
+        if not self.options.allow_libiconv:
```

Comment:
I am little confused why this needs to be an option... playing with this locally I was able to build this without any modifications to the recipe 🤔 

```
[requires]
opentdf-client/1.1.5

[options]
libxml2:iconv = False
libxml2:icu = False
boost:i18n_backend_iconv = off
boost:i18n_backend_icu = False
boost:without_locale = True
```


[graph.html](https://github.com/conan-io/conan-center-index/files/9481838/graph.html.txt)

Reply:
It's a convenience option for the consumer of the recipe.  Several of the environments that our customers run in do NOT include libiconv, plus the licensing for libiconv is something we want to stay away from.

100% necessary?  No.  Useful and unobtrusive?  Yep.  That's why I added it...to help our package's consumers.

Reply:
First time I see such feature in a recipe I guess. But, as your are the author, I will not argue about. However, if you want to be sure, I would recommend using `validate()` method and iterate over dependencies to check if, somehow, `libiconv` is there:

```python
def validate(self):
    try:
        if self.dependencies["libiconv"] and not self.options.allow_libiconv:
            self.output.warn(f"libiconv has been found in dependencies tree along the option {self.name}:allow_libiconv=False")
    except KeyError:
        pass
```

Reply:
OK, that's a good idea in addition to the changes I have in there already.  I will add that...thanks!

Reply:
Added...